database/actions.go

```diff
@@ -158,6 +202,19 @@ func (db *DB) InsertAction(a mig.Action) (err error) {
 	if err != nil {
 		return fmt.Errorf("Failed to store action: '%v'", err)
 	}
+
+	// The following query establishes a relation between the new action and all
+	// agents targeted by the action.
+	_, err = db.c.Exec(fmt.Sprintf(`
+  insert into agent_action_relation (agent_id, action_id)
+  select $1, A.id
+  from agents A
+  where (%s);
```

Comment:
isn't that a sql injection?

Reply:
I'm an idiot... of course it's a sql injection, that's what the target is in the first place.
Sorry about that, it's been years since I looked at that code